webui/react/src/pages/F_ExpList/glide-table/MultiSortMenu.tsx

```diff
@@ -0,0 +1,197 @@
+import { Popover } from 'antd';
+
+import Button from 'components/kit/Button';
+import Select from 'components/kit/Select';
+import { V1ColumnType, V1ProjectColumn } from 'services/api-ts-sdk';
+import Icon from 'shared/components/Icon';
+import { Loadable } from 'utils/loadable';
+
+import css from './MultiSortMenu.module.scss';
+
+const INITIAL_SORTS = [{ column: undefined, direction: undefined }];
+
+type DirectionType = 'asc' | 'desc';
+export interface Sort {
+  column?: string;
+  direction?: DirectionType;
+}
+export type ValidSort = Required<Sort>;
+
+export const isValidSort = (s: Sort): s is ValidSort => !!(s.column && s.direction);
```

Comment:
nit: I understand that `!!` basically turns a truthy value into `true`, but is it needed when you're already doing boolean logic (`&&`)?

Reply:
Yes -- boolean operations don't convert their operands to booleans:
```js
(undefined && "asc") === undefined // true
("validation.limit" && undefined) === undefined // also true
```
The return type of the function must be a boolean in order to be a type guard.